### ISSUE TITLE
req aborted error

### ISSUE THREAD

**Issue by Sriram-27727** [balderdashy/skipper-s3#40]
i am using skipper-s3(0.5.5) in sails js application. I am getting an issue (mentioned below) while file uploading. i got this error sometimesonly, not all the times, due to this error my server crashes every time. I couldn't get why this error is coming.

events.js:72
        throw er; // Unhandled 'error' event
              ^
Error: Request aborted
    at IncomingMessage.onReqAborted (/home/ubuntu/workspace/zugphonics-server/node_modules/sails/node_modules/skipper/node_modules/multiparty/index.js:175:17)
    at IncomingMessage.EventEmitter.emit (events.js:92:17)
    at abortIncoming (http.js:1911:11)
    at Socket.serverSocketCloseListener (http.js:1923:5)
    at Socket.EventEmitter.emit (events.js:117:20)
    at TCP.close (net.js:466:12)

Please help me in this.

**Comment by mattcodez** [comment]
I'm currently struggling with the same issue. Odd thing is, I only have a problem in production. On my local development machine, it works fine. @Sriram-27727 Did you ever find a solution?

**Comment by mattcodez** [comment]
@Sriram-27727 I was able to at least catch the error by wrapping the upload call in a domain. Idea taken from here: https://github.com/balderdashy/skipper/issues/49#issuecomment-130401082

**Comment by boshido** [comment]
I found the same problem.
Anyway @mattcodez  thank you for sharing this.

```Unhandled rejection Error: Request aborted
    at IncomingMessage.onReqAborted (/var/source/node_modules/skipper/node_modules/multiparty/index.js:175:17)
    at IncomingMessage.emit (events.js:92:17)
    at abortIncoming (http.js:1912:11)
    at CleartextStream.serverSocketCloseListener (http.js:1924:5)
    at CleartextStream.emit (events.js:117:20)
    at tls.js:694:10
    at /var/source/node_modules/async-listener/glue.js:188:31
    at process._tickDomainCallback [as _tickCallback] (node.js:492:13)```

**Comment by sailsbot** [comment]
Thanks for posting, @Sriram-27727.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by boshido** [comment]
Don't close this issue 
The bug still appears

**Comment by yaK2manD** [comment]
Yes. Had the same.

**Comment by sgress454** [comment]
Please see https://github.com/balderdashy/skipper/issues/49.  It would be helpful if you could test against the updated Skipper and see if the problem is resolved for you:

```
npm install github://balderdashy/skipper
```

and then in config/http.js:

```
module.exports.http = {
  bodyParser: require('skipper'),
  ...rest of file...
}
```

**Comment by karuna142** [comment]
> @Sriram-27727 I was able to at least catch the error by wrapping the upload call in a domain. Idea taken from here: [balderdashy/skipper#49 (comment)](https://github.com/balderdashy/skipper/issues/49#issuecomment-130401082)

can you plz share how you have handled the error?

**Comment by chandanpal** [comment]
> @Sriram-27727 I was able to at least catch the error by wrapping the upload call in a domain. Idea taken from here: [balderdashy/skipper#49 (comment)](https://github.com/balderdashy/skipper/issues/49#issuecomment-130401082)

can you please share, how can we handle the  aborted upload error. I have used domain but still getting same issue.

    at IncomingMessage.onReqAborted (/home/spyglass/callhome_from_pod/callhome/node_modules/skipper/node_modules/multiparty/index.js:175:17)
    at IncomingMessage.emit (events.js:182:13)
    at IncomingMessage.EventEmitter.emit (domain.js:442:20)
    at abortIncoming (_http_server.js:444:9)
    at socketOnClose (_http_server.js:437:3)
    at Socket.emit (events.js:187:15)
    at Socket.EventEmitter.emit (domain.js:442:20)
    at TCP._handle.close [as _onclose] (net.js:598:12)
